xpm/xpm.go

```diff
@@ -72,10 +75,10 @@ func parseDimensions(data string) (w, h, i, j int) {
 		return
 	}
 
-	w, _ = strconv.Atoi(parts[0])
-	h, _ = strconv.Atoi(parts[1])
-	i, _ = strconv.Atoi(parts[2])
-	j, _ = strconv.Atoi(parts[3])
+	w, err = strconv.Atoi(parts[0])
+	h, err = strconv.Atoi(parts[1])
+	i, err = strconv.Atoi(parts[2])
+	j, err = strconv.Atoi(parts[3])
```

Comment:
I am not certain here, but wouldn't this code only return an error if it happen during the parsing of parts[3]?

Reply:
I was trying to be clever to avoid error handling code twice as long as the real code :(.
Added the returns.